pkg/envtest/envtest_test.go

```diff
@@ -226,8 +226,228 @@ var _ = Describe("Test", func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(crd.Spec.Names.Kind).To(Equal("Driver"))
+
+			err = WaitForCRDs(env.Config, []*v1beta1.CustomResourceDefinition{
+				{
+					Spec: v1beta1.CustomResourceDefinitionSpec{
+						Group:   "qux.example.com",
+						Version: "v1beta1",
+						Names: v1beta1.CustomResourceDefinitionNames{
+							Plural: "bazs",
+						}},
+				},
+				{
+					Spec: v1beta1.CustomResourceDefinitionSpec{
+						Group:   "bar.example.com",
+						Version: "v1beta1",
+						Names: v1beta1.CustomResourceDefinitionNames{
+							Plural: "foos",
+						}},
+				},
+				{
+					Spec: v1beta1.CustomResourceDefinitionSpec{
+						Group:   "crew.example.com",
+						Version: "v1beta1",
+						Names: v1beta1.CustomResourceDefinitionNames{
+							Plural: "captains",
+						}},
+				},
+				{
+					Spec: v1beta1.CustomResourceDefinitionSpec{
+						Group:   "crew.example.com",
+						Version: "v1beta1",
+						Names: v1beta1.CustomResourceDefinitionNames{
+							Plural: "firstmates",
+						}},
+				},
+				{
+					Spec: v1beta1.CustomResourceDefinitionSpec{
+						Group: "crew.example.com",
+						Names: v1beta1.CustomResourceDefinitionNames{
+							Plural: "drivers",
+						},
+						Versions: []v1beta1.CustomResourceDefinitionVersion{
+							{
+								Name:    "v1",
+								Storage: true,
+								Served:  true,
+							},
+							{
+								Name:    "v2",
+								Storage: false,
+								Served:  true,
+							},
+						}},
+				},
+			},
+				CRDInstallOptions{MaxTime: 50 * time.Millisecond, PollInterval: 15 * time.Millisecond},
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			close(done)
+		}, 5)
 	})
 
+	It("should update CRDs if already present in the cluster", func(done Done) {
+
+		// Install only the CRDv1 multi-version example
+		crds, err = InstallCRDs(env.Config, CRDInstallOptions{
+			Paths: []string{filepath.Join(".", "testdata", "crdv1_original")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect to find the CRDs
+
+		crd := &v1beta1.CustomResourceDefinition{}
+		err = c.Get(context.TODO(), types.NamespacedName{Name: "drivers.crew.example.com"}, crd)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crd.Spec.Names.Kind).To(Equal("Driver"))
+		Expect(len(crd.Spec.Versions)).To(BeEquivalentTo(2))
+
+		// Store resource version for comparison later on
+		firstRV := crd.ResourceVersion
+
+		// Add one more version and update
+		crds, err = InstallCRDs(env.Config, CRDInstallOptions{
+			Paths: []string{filepath.Join(".", "testdata", "crdv1_updated")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Expect to find updated CRD
+
+		crd = &v1beta1.CustomResourceDefinition{}
+		err = c.Get(context.TODO(), types.NamespacedName{Name: "drivers.crew.example.com"}, crd)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(crd.Spec.Names.Kind).To(Equal("Driver"))
+		Expect(len(crd.Spec.Versions)).To(BeEquivalentTo(3))
+		Expect(crd.ResourceVersion).NotTo(BeEquivalentTo(firstRV))
```

Comment:
you'll also need a `close(done)` at the end here, or just remove the done argument and the timeout

Reply:
thank you for putting me out of my misery 🙇